docs/migration/dapp_migration.md

```diff
@@ -0,0 +1,46 @@
+- **Calldata compression 🥳**: Compression takes place protocol level; dapps don't need to change anything, data will just get cheaper! (You are charged even less if your calldata is highly compressible with brotli.)
+- **Support for All Ethereum L1 precompiles 🥳**: (`blake2f`, `ripemd160`, etc)
+- **Tighter Syncronization with L1 Block Numbers / Timestamps 🥳**:  L1 block number and timestamps (accessed via `block.number` and `block.timestamp` on L2) are updated more frequently in Nitro than in Arbitrum classic; expect them to be nearly real-time/ in sync with L1. 
+
+- **L2 Block hash EVM Consistency 🥳**: L2 block hashes take the same format as on Ethereum (if you query it from the ArbSys precompile, not the one in `block.hash(uin256)`).
+
+
+- **Geth tracing 🥳**: `debug_traceTransaction` RPC endpoint is supported; this includes tracing of ArbOS internal bookkeeping actions.
+
+## Breaking changes
+
+#### Dapps
+- **Gas Accounting:** Since gas accounting is now consistent with the L1 EVM, and hard-coded gas values (used, e.g. in conjuction with `gasLeft`) should be changed accordingly.
+- **Retryable Tickets**: 
+    - The submission cost is now collected/enforced in the L1 inbox and checked against the L1 transaction's `msg.value`; contracts cannot/should not rely on funds pooled in the L2 destination to cover this cost.
+    - For the redemption of retryable tickets, the calculation of the L2 transaction ID is changed, as has the transaction lifecycle of attempting multiple redemptions (i.e., after failed attempts). See [arbitrum-sdk](https://github.com/OffchainLabs/arbitrum-sdk/blob/fe3c3ee90a2d713955988dcb6a9f87732b7dbedc/src/lib/message/L1ToL2Message.ts#L547) for the new client-side flow (detailed documentation coming soon) 
+
+#### Protocol Contracts 
+
+- **New Contract Deployments**: For the Nitro upgrade, the following contracts will be redeployed on L1 to new addresses:
+    - SequencerInbox
+    - RollupCore
+    - Outbox
+
+- **Sequencer Inbox changes**: The Sequencer inbox has a new interface and requires a new approach to determining a transaction's inclusion on L1; see [arbitrum-sdk](https://github.com/OffchainLabs/arbitrum-sdk/blob/fe3c3ee90a2d713955988dcb6a9f87732b7dbedc/src/lib/message/L2Transaction.ts#L165) for client-side flow. 
+
+
+- **Outbox Changes**: The Outbox has a new (simplified!) architecture; in short, all outgoing messages will be included in a single Merkle tree (vs Arbitrum classic, in which many outbox entries, each with its own Merkle root, get created over time). See [arbitrum-sdk](https://github.com/OffchainLabs/arbitrum-sdk/blob/fe3c3ee90a2d713955988dcb6a9f87732b7dbedc/src/lib/message/L2ToL1Message.ts#L479) for new flow of interacting with the outbox, which includes interactions with new methods in the NodeInterface virtual precompile (detailed documentation coming soon).
+
+#### Other
+- **No Parity tracing**: Initially `trace_filter` RPCs won't be available; they will be in the coming months.
```

Comment:
Maybe mention that as a replacement the geth tracing APIs are available (I know it was mentioned earlier, but mentioning it again might avoid confusion)